### PR TITLE
Coloursensor

### PR DESCRIPTION
**Pull request by WhyNot180**
# What is it?

This is the colour sensor code integrated into the code already in the main branch.

## How does it work?

**_(Note that the library uses the American spelling of colour)_**

First, we must declare the I2C port (on board or MXP).
Then, we have declarations for the colour sensor, colour matcher, colour values for red, green, yellow, and blue (constants), and the booleans for telling the roborio which colour the ball is plus an additional emergency dump bool in case the colour sensor detects the wrong colour (Ideally the operator will be able to press a button to override the sensor and tell the robot to dump the ball).
Next, in the initialization we need to add the colours to the colour matcher.
Once the robot has been initialized, the code matches the closest colour and reports it to the dashboard along with the colour values (RGB).

Normally, I'd make this fancier and add a more thorough explanation, but I have homework to do.

### CODE REVIEW

File: src/main/java/frc/robot/Robot.java
Comment (theyeeman) [inline src/main/java/frc/robot/Robot.java]:
```suggestion
      ballIsBlue = false;
      ballIsRed = true;
```

Nit: switch the order of these two variables so it's the same as all the other `else if` statements. Makes it slightly easier to read the code.

FYI: "Nit" stands for "nitpick" and it means that the change I am suggesting isn't necessary (it's more of a preference type of thing). It just gives you some ideas on how things would have been done by other collaborators. Don't feel like you need to agree with my suggestion though! 😄

File: src/main/java/frc/robot/Robot.java
Comment (theyeeman) [inline src/main/java/frc/robot/Robot.java:49]: Nit: is it possible to give this color sensor a better variable name instead of `colorSensor1`? Maybe if we are planning to have multiple color sensors connected, then we can create an ArrayList and access each color sensor with an index like `colorSensors[1]`?

Any thoughts/opinions on this?

Comment (UserC2) [reply src/main/java/frc/robot/Robot.java:49]: Each colour sensor will probably have a separate purpose, so we could name it based on its use.

I think that would be less complex than creating an array of them.

File: src/main/java/frc/robot/Robot.java
Comment (theyeeman) [inline src/main/java/frc/robot/Robot.java:182]: Is `dumpBall` just a placeholder for now and isn't being used?

File: src/main/java/frc/robot/Robot.java
Comment (theyeeman) [reply src/main/java/frc/robot/Robot.java:49]: Sounds good to me! Let's just leave it as `colorSensor1` for now then.

File: src/main/java/frc/robot/Robot.java
Comment (theyeeman) [inline src/main/java/frc/robot/Robot.java]: I want to make sure that the changes to the methods `disabledInit()` and `disabledPeriodic()` are intentional? I remember @newjanson was making changes here, so I want to make sure that this is okay.

Comment (UserC2) [reply src/main/java/frc/robot/Robot.java]: That should probably remain as it was before, its for the bar PID. I don't think it's supposed to be deleted.

File: src/main/java/frc/robot/Robot.java
Comment (WhyNot180) [reply src/main/java/frc/robot/Robot.java:182]: Yeah, it's a placeholder until we get the intake and turret finished

File: src/main/java/frc/robot/Robot.java
Comment (WhyNot180) [reply src/main/java/frc/robot/Robot.java]: Yeah, he set it to coast mode on saturday so it's part of the original code

Comment (theyeeman) [reply src/main/java/frc/robot/Robot.java]: So the changes in the PR are good?

Comment (WhyNot180) [reply src/main/java/frc/robot/Robot.java]: Disabled init should have hookMotor.setIdleMode(IdleMode.kCoast)

Comment (theyeeman) [reply src/main/java/frc/robot/Robot.java]: Can you make the changes for that @WhyNot180? And can you also double check `disabledPeriodic()` to make sure that change was intentional?

Comment (WhyNot180) [reply src/main/java/frc/robot/Robot.java]: Okay, I fixed it and pushed it

**Review by theyeeman — APPROVED** [review]

File: src/main/java/frc/robot/Robot.java
Comment (theyeeman) [reply src/main/java/frc/robot/Robot.java]: Thank you! I'm going to wait for Greg to review it before merging

**Pull request merged**
